### PR TITLE
Public study api

### PR DESCRIPTION
**Pull request by alxdarksage**
Adding a single endpoint to retrieve a partial study config that does not require authentication, and which is cached.

BRIDGE-3046.

### CODE REVIEW

**Comment by alxdarksage** [comment]
Create an API for any signed in user to get a complete study record.

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudy.java [hunk 1]
```diff
@@ -118,18 +118,19 @@ public HibernateStudy() {}
...
         this.createdOn = createdOn;
         this.modifiedOn = modifiedOn;
-        this.deleted = deleted;
-        this.phase = phase;
-        this.studyLogoUrl = logoURL;
-        this.version = version;
+        this.studyLogoUrl = studyLogoUrl;
+        this.colorScheme = colorScheme;
+        this.signInTypes = signInTypes;
     }
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudy.java:133]: Revert this.

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudyDao.java [hunk 2]
```diff
@@ -24,8 +24,9 @@
     
     static final String COUNT_PHRASE = "select count(*) ";
     static final String SELECT_PHRASE = "select new org.sagebionetworks.bridge.hibernate."
-            + "HibernateStudy(study.name, study.identifier, study.appId, study.createdOn, "
-            + "study.modifiedOn, study.deleted, study.phase, study.studyLogoUrl, study.version) ";
+            + "HibernateStudy(study.name, study.identifier, study.details, study.phase, "
+            + "study.deleted, study.createdOn, study.modifiedOn, study.studyLogoUrl, "
+            + "study.colorScheme, study.signInTypes) ";
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudyDao.java]: revert

File: src/test/java/org/sagebionetworks/bridge/hibernate/HibernateStudyTest.java [hunk 3]
```diff
@@ -42,17 +44,19 @@
...
+        assertEquals(study.getDetails(), "details");
         assertEquals(study.getCreatedOn(), CREATED_ON);
         assertEquals(study.getModifiedOn(), MODIFIED_ON);
         assertTrue(study.isDeleted());
+        assertEquals(study.getColorScheme(), COLOR_SCHEME);
         assertEquals(study.getPhase(), DESIGN);
         assertEquals(study.getStudyLogoUrl(), TEST_LINK);
-        assertEquals(study.getVersion(), Long.valueOf(10L));
+        assertEquals(study.getSignInTypes(), ImmutableList.of(EMAIL_MESSAGE));
     }
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/hibernate/HibernateStudyTest.java]: Revert constructor test

File: src/test/java/org/sagebionetworks/bridge/hibernate/HibernateStudyTest.java [hunk 4]
```diff
@@ -177,4 +149,64 @@ public void canSerialize() throws Exception {
...
+        
+        JsonNode summary = BridgeObjectMapper.get().readTree(
+                Study.STUDY_SUMMARY_WRITER.writeValueAsString(study));
+        assertEquals(summary.size(), 10);
+        assertEquals(summary.get("name").textValue(), "name");
+        assertEquals(summary.get("identifier").textValue(), "oneId");
+        assertEquals(summary.get("details").textValue(), "someDetails");
+        assertEquals(summary.get("phase").textValue(), "analysis");
+        assertEquals(summary.get("createdOn").textValue(), CREATED_ON.toString());
+        assertEquals(summary.get("modifiedOn").textValue(), MODIFIED_ON.toString());
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/hibernate/HibernateStudyTest.java]: Remove createdOn & modifiedOn too.

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyControllerTest.java [hunk 5]
```diff
@@ -341,4 +399,47 @@ public void finishStudyLogo_noFinishedRevision() {
...
+        verify(mockCacheProvider).setObject(key, json, ONE_DAY_IN_SECONDS);
+    }
+    
+    @Test
+    public void getStudyForApp_cached() throws JsonProcessingException {
+        Study study = Study.create();
+        study.setName("Name1");
+        study.setIdentifier("id1");
+        study.setVersion(10L);
+        when(service.getStudy(TEST_APP_ID, TEST_STUDY_ID, true)).thenReturn(study);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyControllerTest.java]: You could probably remove this...

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyControllerTest.java [hunk 6]
```diff
@@ -341,4 +399,47 @@ public void finishStudyLogo_noFinishedRevision() {
...
+        when(mockCacheProvider.getObject(key, String.class)).thenReturn(json);
+        
+        String retValue = controller.getStudyForApp(TEST_APP_ID, TEST_STUDY_ID);
+        
+        Study deser = BridgeObjectMapper.get().readValue(retValue, Study.class);
+        assertEquals(deser.getName(), "Name1");
+        assertEquals(deser.getIdentifier(), "id1");
+        assertNull(deser.getVersion());
+        
+        verify(mockCacheProvider, never()).setObject(key, json, ONE_DAY_IN_SECONDS);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyControllerTest.java]: Verify it never calls getStudy()

**Review by jnelsonjava — APPROVED** [review]

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudy.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateStudy.java:133]: reverted

**Pull request merged**